### DISCUSSION TITLE
zfs-windows-2.2.6-rc5

### DISCUSSION THREAD

**Discussion by lundman** [openzfsonwindows/openzfs#400] [Development]
I posted the rc5 build, it deals mostly with snapshot mounting BSODs.

Please report your findings here, if your bugs are still active, can you mention the bug ID in here for me thanks.

**Comment by lundman** [comment]
OK changing driveletter from E: to Y: (ie, from driveletter to a different driveletter) does not work. Code sees it is a driveletter and is happy. It does not check it is the desired driveletter

**Comment by lundman** [comment]
Like an idiot, I updated VisualStudio and now I can't deploy - which makes debugging a pain.  I've informed MS, but meanwhile, I have to load it manually which makes me stabby. :)

**Reply by sskras** [reply]
If you work in a VM, you probably didn't make a snapshot before the upgrade, right?

**Reply by lundman** [reply]
TargetVM I always use snapshots, so that didn't change. HostVM I do not. Been trying many things, including downloading the latest Windows Development VM, and configuring remote target. Also errors. The latest versions are just broken.

**Reply by sskras** [reply]
Yes, I meant HostVM (AKA BuildVM).

**Reply by lundman** [reply]
I tried two brand newly downloaded Windows Development VMs, and they can't deploy either, so they just broke it. Anyway, I have a manual way to deploy for now, it's not great but works. At least we are still moving forwards.

**Reply by lundman** [reply]
OK starting to look good, assigning marginal trust to new mount code. 
```
 $ out/build/x64-Debug/cmd/zpool/zpool.exe create -O casesensitivity=insensitive BOOM PHYSICALDRIVE0
working on dev '#1048576#10726932480#\\?\PHYSICALDRIVE0'
setting path here '/dev/physicaldrive0'
setting physpath here '#1048576#10726932480#\\?\PHYSICALDRIVE0'
Expanded path to '\\?\PHYSICALDRIVE0'

$ out/build/x64-Debug/cmd/zfs/zfs set driveletter=Y BOOM
zunmount(BOOM,E:/) running
zunmount(BOOM,E:/) returns 0

$ out/build/x64-Debug/cmd/zfs/zfs create BOOM/lower                             

$ out/build/x64-Debug/cmd/zfs/zfs snap BOOM@send                                
$ out/build/x64-Debug/cmd/zfs/zfs mount BOOM@send                               
ZFS: snapshot mountpoint 'Y://.zfs/snapshot/send/'

$ out/build/x64-Debug/cmd/zfs/zfs mount
BOOM                            Y:/
BOOM/lower                      Y://lower
BOOM@send                       Y://.zfs/snapshot/send/

$ out/build/x64-Debug/cmd/zpool/zpool.exe export BOOM
zunmount(BOOM/lower,Y://lower) running
zunmount(BOOM/lower,Y://lower) returns 0
zunmount(BOOM@send,Y://.zfs/snapshot/send/) running
zunmount(BOOM@send,Y://.zfs/snapshot/send/) returns 0
zunmount(BOOM,Y:/) running
zunmount(BOOM,Y:/) returns 0
```

No leaks, no forced closed vnodes. I'm sure there are corner cases but a good start
